component/src/main/java/org/ballerinalang/net/kafka/nativeimpl/functions/consumer/GetAssignment.java

```diff
@@ -0,0 +1,101 @@
+import org.ballerinalang.net.kafka.Constants;
+import org.ballerinalang.util.codegen.PackageInfo;
+import org.ballerinalang.util.codegen.StructInfo;
+import org.ballerinalang.util.exceptions.BallerinaException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Native function ballerina.net.kafka:getAssignment which returns given partition assignment for consumer.
+ */
+@BallerinaFunction(packageName = "ballerina.net.kafka",
+        functionName = "getAssignment",
+        receiver = @Receiver(type = TypeKind.STRUCT, structType = "KafkaConsumer",
+                structPackage = "ballerina.net.kafka"),
+        args = {
+                @Argument(name = "c",
+                        type = TypeKind.STRUCT, structType = "KafkaConsumer",
+                        structPackage = "ballerina.net.kafka")
+        },
+        returnType = {@ReturnType(type = TypeKind.ARRAY, elementType = TypeKind.STRUCT, structType = "TopicPartition",
+                structPackage = "ballerina.net.kafka"),
+                @ReturnType(type = TypeKind.STRUCT)},
+        isPublic = true)
+public class GetAssignment extends AbstractNativeFunction {
+
+    @Override
+    public BValue[] execute(Context context) {
+
+        BStruct consumerStruct = (BStruct) getRefArgument(context, 0);
+
+        KafkaConsumer<byte[], byte[]> kafkaConsumer = (KafkaConsumer) consumerStruct
+                .getNativeData(Constants.NATIVE_CONSUMER);
+        if (kafkaConsumer == null) {
+            throw new BallerinaException("Kafka Consumer has not been initialized properly.");
+        }
+
+        try {
+            Set<TopicPartition> assignments = kafkaConsumer.assignment();
+            List<BStruct> assignmentList = new ArrayList<>();
+            if (!assignments.isEmpty()) {
+                assignments.forEach(assignment -> {
+                    BStruct infoStruct = createPartitionStruct(context);
+                    infoStruct.setStringField(0, assignment.topic());
+                    infoStruct.setIntField(0, assignment.partition());
+                    assignmentList.add(infoStruct);
+                });
+            }
+            return getBValues(new BRefValueArray(assignmentList.toArray(new BRefType[0]),
+                    createPartitionStruct(context).getType()));
+        } catch (KafkaException e) {
+            return getBValues(null, BLangVMErrors.createError(context, 0, e.getMessage()));
+        }
+    }
+
+    private BStruct createPartitionStruct(Context context) {
+        PackageInfo kafkaPackageInfo = context.getProgramFile()
+                .getPackageInfo(Constants.KAFKA_NATIVE_PACKAGE);
+        StructInfo consumerRecordStructInfo = kafkaPackageInfo
+                .getStructInfo(Constants.TOPIC_PARTITION_STRUCT_NAME);
+        BStructType structType = consumerRecordStructInfo.getType();
+        BStruct bStruct = new BStruct(structType);
+        return bStruct;
+    }
```

Comment:
This method is duplicated on several native function implementation classes. Shall we move them to a common place like a util class or parent class

Reply:
+1